contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,397 @@
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor (address dai, address eth) {
+    // [todo] Move to intiialize function
+    // The minimum cover premium is 2.6%. 20% of the cover premium is: 2.6% * 20% = 0.52%
+    CONFIG.REWARD_PERC = 52;
+
+    CONFIG.INCIDENT_IMPACT_ESTIMATE_PERC = 30; // 30%
+    CONFIG.MIN_VOTING_PERIOD_DAYS = 3; // days
+    CONFIG.MAX_VOTING_PERIOD_DAYS = 30; // days
+    CONFIG.PAYOUT_COOLDOWN_DAYS = 1; //days
+    CONFIG.CLAIM_ASSESSMENT_DEPOSIT_PERC = 500; // 5% i.e. 0.05 ETH submission flat fee
+    CONFIG.INCIDENT_ASSESSMENT_DEPOSIT_PERC = 0;
+    addressOfAsset[uint(Asset.ETH)] = eth;
+    addressOfAsset[uint(Asset.DAI)] = dai;
+
+    nxm = INXMToken(master.tokenAddress());
+  }
+
+  /* ========== VIEWS ========== */
+
+  function tokenController() internal view returns (ITokenController) {
+    return ITokenController(getInternalContractAddress(ID.TC));
+  }
+
+  function memberRoles() internal view returns (IMemberRoles) {
+    return IMemberRoles(getInternalContractAddress(ID.MR));
+  }
+
+  function pool() internal view returns (IPool) {
+    return IPool(getInternalContractAddress(ID.P1));
+  }
+
+  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
+    return votesOf[assessor].length;
+  }
+
+  function getClaimsCount() external override view returns (uint) {
+    return claims.length;
+  }
+
+  function getIncidentsCount() external override view returns (uint) {
+    return incidents.length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param withProof        When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if withProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool withProof,
+    string calldata ipfsProofHash
+  ) external payable onlyMember {
+    AssessmentClaimsLib.submitClaim(
+      CONFIG,
+      coverId,
+      requestedAmount,
+      withProof,
+      ipfsProofHash,
+      claims
+    );
+
+  }
+
+  function submitIncident(
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external {
+    (
+      AffectedToken memory affectedToken,
+      Incident memory incident
+    ) = AssessmentIncidentsLib.getIncidentToSubmit(
+      CONFIG,
+      nxm,
+      memberRoles(),
+      productId,
+      priceBefore,
+      date
+    );
+
+    AssessmentIncidentsLib.saveIncident (
```

Comment:
this function has about 4 lines of code is it possible to inline here? how close are we to max limit of size